extensions/classify/config.proto

```diff
@@ -0,0 +1,98 @@
+/* Copyright 2020 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package istioclassify;
+
+// Configuration to classify traffic.
+message PluginConfig {
+  // next id: 2
+  // Multiple independent classification configurations.
+  repeated Classification classifications = 1;
+}
+
+// Classification uses [well known attributes](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/rbac_filter#condition) as inputs and produces
+// a new attribute that denotes a specific classification.
+// The new attribute can be used by a downstream plugin.
+//
+// Example: A classifier may produce an attribute called `istio.operationId` which is analogous to the `operationId` in the OpenAPI Spec.
+// The Stats plugin will run after the classifier and use `istio.operationId` to populate a dimension. 
+message Classification {
+  // Phase denotes plugin lifecycle phase when the classification is evaluated. 
+  Phase phase = 1;
+
+  // The following attribute is populated on successfully running the classification.
+  // Example: `istio.operationId`
+  string output_attribute = 2;
+
+  // A Classification may fail to evaluate when an attribute is not available.
+  // Example: `response.code` may not be available when a request ends abruptly.
+  // When evalution failes, the plugin will return an error to the client.
+  // This is the correct behaviour if the result of the classification is used by an authz plugin.
+  // If the result of the classifier is used by stats, you may just want to log an error but not
+  // fail the request. If `ignore_errors=true` then the classifier plugin will not fail the request.
+  bool ignore_errors = 3;
```

Comment:
This is interesting and can have broad security implications if any security features are dependent on these attributes.

Should this be always set to "false" and not even exposed as a field i.e. "attribute" is best effort with defaults. Downstream filters should be created in a way that they handle unset/default/actual value.

Reply:
If possible, i'd prefer to have some distinction between failure and unset. maybe `istio.attributeGen.status` should be an attribute ;p.

Reply:
This exactly gives us the difference between failure to evaluate and evaluated to empty.
We could put the onus of failing the request of the consumer of the attribute and not the producer.